Kickstarter-iOS/DataSources/BackerDashboardPagesDataSource.swift

```diff
@@ -0,0 +1,57 @@
+import KsApi
+import Library
+import UIKit
+
+internal final class BackerDashboardPagesDataSource: NSObject, UIPageViewControllerDataSource {
+  private let viewControllers: [UIViewController]
+
+  internal init(delegate: UIViewController, sort: DiscoveryParams.Sort) {
+    let backedController = BackerDashboardProjectsViewController()
+    let savedController = BackerDashboardProjectsViewController()
+
+    if let delegate = delegate as? BackerDashboardProjectsViewControllerDelegate {
+      backedController.configureWith(delegate: delegate, projectsType: .backed, sort: sort)
+      savedController.configureWith(delegate: delegate, projectsType: .saved, sort: sort)
+    }
+
+    var controllers: [UIViewController] = []
+    controllers[BackerDashboardTab.backed.rawValue] = backedController
```

Comment:
this is a bit dangerous since if we ever changed the order of the enum cases we would get a crash when we tried doing `controllers[1] = ...` before `controllers[0] = ...`.

rather than the order being determined by the order of the enum cases (which is fragile and doesnt play nicely with our alphabetizing fanaticism!), we could create a static array that determines the order for us:

```swift
case BackerDashboardTab {
  case backed
  case saved

  public static allTabs: [BackerDashboardTab] = [.backed, .saved]
}
```

and then anywhere we need to do something with the tabs that depends on the order we could do something like:

```swift
self.viewControllers = BackerDashboardTab.allTabs.map { tab in 
  switch tab {
  case .backed: return backedController
  case .saved: return savedController
  }
}
```

added bonus of this approach is that adding a new tab will cause compiler errors everywhere so we can correct it for free!